.github/workflows/ci.yaml

```diff
@@ -86,60 +51,156 @@ jobs:
       #      https://api.github.com/repos/${{ github.actor }}/${{ github.repository }}/actions/runs/${{ github.run_id }}/timing
 
+
+  build_job:
+    timeout-minutes: 30
+    runs-on: ubuntu-20.04
+    needs: test_job
+
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+
+    permissions:      
+      contents: read  #permissions for publishing Docker images to Github Packages
+      packages: write #permissions for publishing Docker images to Github Packages
+    
+    defaults:
+      run:
+        working-directory: angular
+    
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Buld phase
+        run: npm run build
+
+      #---Build Docker image, Login & Publish to Docker Packages---
+      - name: Log in to the Container registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login $REGISTRY -u ${{ github.actor }} --password-stdin
+
+      - name: build Docker image on push event
+        if: github.event_name == 'push'
+        run: docker build -t $REGISTRY/$IMAGE_NAME:${GITHUB_SHA::7} -t $REGISTRY/$IMAGE_NAME:latest .
+
+      - name: build Docker image on PR event
+        if: github.event_name == 'pull_request'
+        run: |
+          PR_SHA7=${{github.event.pull_request.head.sha}}
+          docker build -t $REGISTRY/$IMAGE_NAME:${PR_SHA7::7} -t $REGISTRY/$IMAGE_NAME:latest .
+
+      - name: push Docker image on push event
+        if: github.event_name == 'push'
+        run: docker push $REGISTRY/$IMAGE_NAME:${GITHUB_SHA::7} && docker push $REGISTRY/$IMAGE_NAME:latest
+
+      - name: push Docker image on PR event
+        if: github.event_name == 'pull_request'
+        run: |
+          PR_SHA7=${{github.event.pull_request.head.sha}}
+          docker push $REGISTRY/$IMAGE_NAME:${PR_SHA7::7} && docker push $REGISTRY/$IMAGE_NAME:latest
+
+      #---Build Docker image, Login & Publish to Docker Packages - WITH 3RD PARTY ACTIONS---
+      #- name: Log in to the Container registry
+      #  uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+      #  with:
+      #    registry: $REGISTRY
+      #    username: ${{ github.actor }} #${{ github.repository_owner }}
+      #    password: ${{ secrets.GITHUB_TOKEN }}
       
+      #- name: Extract metadata (tags, labels) for Docker
+      #  id: meta
+      #  uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+      #  with:
+      #    images: $REGISTRY/$IMAGE_NAME
+            
+      #- name: Build and push Docker image
+      #  uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+      #  with:
+      #    context: ./angular
+      #    push: true
+      #    tags: ${{ steps.meta.outputs.tags }}
+      #    labels: ${{ steps.meta.outputs.labels }}
```

Comment:
don't keep the commented out code in the repo.

Reply:
ok